js/components/thumbnailCarousel.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * ThumbnailCarousel component
+ *
+ * @module ThumbnailCarousel
+ */
+
+var React = require('react'),
+    ReactDOM = require('react-dom'),
+    Utils = require('./utils');
+
+var ThumbnailCarousel = React.createClass({
+  getInitialState: function() {
+    return {
+      thumbnailWidth: 0,
+      thumbnailHeight: 0,
+      centerThumbnailWidth: 0,
+      centerThumbnailHeight: 0
+    };
+  },
+
+  componentDidMount: function() {
+    var thumbnail = ReactDOM.findDOMNode(this.refs.thumbnail);
+    var carousel = ReactDOM.findDOMNode(this.refs.thumbnailCarousel);
+    this.setState({thumbnailWidth: thumbnail.clientWidth,
+                   thumbnailHeight: thumbnail.clientHeight,
+                   centerThumbnailWidth: carousel.clientWidth,
+                   centerThumbnailHeight: carousel.clientHeight});
+  },
+
+  shouldComponentUpdate: function(nextProps) {
+    return (nextProps.hoverPosition != this.props.hoverPosition);
```

Comment:
the `ThumbnailCarousel` component isn't updating, we need to pass `hoverPosition` as a prop in scrubberBar.js for this to work